packages/web3/src/web3.ts

```diff
@@ -92,10 +92,29 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 					super(jsonInterface, self.getContextObject());
 				}
 
+				self.on(Web3ConfigEvent.CONFIG_CHANGE, event => {
+					for (const contract of ContractBuilder._contracts) {
+						contract.emit(Web3ConfigEvent.CONFIG_CHANGE, event);
+					}
+				});
+
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+				this._eventEmitter.on('ContractCloned', (contract: Contract<any>) => {
+					ContractBuilder._contracts.push(contract);
```

Comment:
contracts list is removed in another PR https://github.com/web3/web3.js/pull/5792/files#diff-55c687488c8bc965581ea1915c7fa86698f5c0d6bf74ae9f41fafc6d0f8281f2L72  , shouldn't we just directly subscribe to change events inside contracts by directly referring super.on and avoid this change in main web3 package

Reply:
done